### PR TITLE
Neo4j backup.service

### PR DESCRIPTION
**Pull request by duffj**

### CODE REVIEW

**Comment by Zapadlo** [comment]
I'm guessing there is an accompanying `coco-provisioner` PR, could you link it here when its raised? Ta

**Comment by duffj** [comment]
@Zapadlo - say what now? Do I have to add something to the `coco-provisioner` repo?

File: neo4j-backup.service [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# remove existing instance
+ExecStartPre=-/usr/bin/docker rm %p
+
+ExecStart=/bin/sh -c "\
+    AWS_ACCESS_KEY=$(/usr/bin/etcdctl get /ft/_credentials/aws/aws_access_key_id);\
+    AWS_SECRET_KEY=$(/usr/bin/etcdctl get /ft/_credentials/aws/aws_secret_access_key); \
+    BUCKET_NAME=$(/usr/bin/etcdctl get /ft/config/neo4j-backup/bucket_name); \
+    DATA_FOLDER=$(/usr/bin/etcdctl get /ft/config/neo4j-backup/data_folder); \
+    TARGET_FOLDER=$(/usr/bin/etcdctl get /ft/config/neo4j-backup/target_folder); \
+    S3_DOMAIN=$(/usr/bin/etcdctl get /ft/config/neo4j-backup/s3_domain); \
```
Comment (Zapadlo) [inline neo4j-backup.service]: L26-29 where will those values come from? re. coco-provisioner changes

Comment (duffj) [inline neo4j-backup.service]: Ah, I added those manually, didn't realise there was an automated way of adding them, thanks.

Comment (duffj) [inline neo4j-backup.service]: As discussed offline, I've modified the source code to use sensible defaults for these parameters so will be removing them from the service file.

**Comment by duffj** [comment]
@Zapadlo - please re-review.

**Comment by Zapadlo** [comment]
Yep, +1

**Comment by duffj** [comment]
@Zapadlo - thanks. Based on the changes in [my other PR](https://github.com/Financial-Times/coco-neo4j-backup/pull/3), I'll do a version bump of the application before merging to `master`.

**Comment by duffj** [comment]
OK, this is ready to go and has been tested extensively in `semantic`, along with a less extensive test in `pre-prod`.

**Comment by scott-ace-newton** [comment]
:+1:

**Comment by duffj** [comment]
Thanks @scott-ace-newton - as discussed with Derek, we're going to do some more testing in pre-prod before merging to `master`, based on the possibility that Amazon might shut the host instance down if it detects excessive resource usage, which is not something we would like to happen unexpectedly in prod.

**Comment by duffj** [comment]
After a conversation with Ammar and @nwrigh today, we have tested the backup and restore processes a number of times in various non-prod environments now, and everything seems pretty good. I'm going to merge this PR.

**Pull request merged**